### PR TITLE
fix(route): 西南石油大学 办公网、教务处、计算机与软件学院、电气信息学院

### PR DESCRIPTION
**Pull request by CYTMWIA**
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例

```routes
/swpu/bgw/zytzgg
/swpu/bgw/bmtzgg
/swpu/bgw/bzhd
/swpu/dean/tzgg
/swpu/dean/xwbd
/swpu/dean/sdsy
/swpu/dxy/1156
/swpu/dxy/1122
/swpu/scs/tzgg
/swpu/scs/xwsd
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [ ] New Route / 新的路由
  - [ ] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [ ] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [ ] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明
更新源数据地址（办公网）；
更新学院名称（计算机科学学院 -> 计算机与软件学院）；
在缓存中保存解析结果，而非`cheerio`对象，避免下次从缓存获取数据时报错；
添加 TS 类型标识；

### CODE REVIEW

File: lib/routes/swpu/dean.ts [hunk 1]
```diff
@@ -44,39 +44,37 @@ async function handler(ctx) {
...
+            async (item) =>
+                (await cache.tryGet(item.link!, async () => {
+                    const resp = await got.get(item.link);
+                    const $ = load(resp.data);
+                    if ($('title').text().startsWith('系统提示')) {
+                        item.author = '系统';
+                        item.description = '无权访问';
+                    } else {
+                        item.author = '教务处';
+                        item.description = $('.v_news_content').html()?.toString();
```
Comment (TonyRL) [inline lib/routes/swpu/dean.ts:66]: [`.html()`](https://cheerio.js.org/docs/api/classes/Cheerio#html) already returned a `string`. No need `toString()` again.

File: lib/routes/swpu/scs.ts [hunk 2]
```diff
@@ -43,45 +43,43 @@ async function handler(ctx) {
...
+            async (item) =>
+                (await cache.tryGet(item.link!, async () => {
+                    const resp = await got.get(item.link);
+                    const $ = load(resp.data);
+                    if ($('title').text().startsWith('系统提示')) {
+                        item.author = '系统';
+                        item.description = '无权访问';
+                    } else {
+                        item.author = '计算机与软件学院';
+                        item.description = $('.v_news_content').html()?.toString();
```
Comment (TonyRL) [inline lib/routes/swpu/scs.ts:66]: [`.html()`](https://cheerio.js.org/docs/api/classes/Cheerio#html) already returned a `string`. No need `toString()` again.

File: lib/routes/swpu/bgw.ts [hunk 3]
```diff
@@ -29,21 +29,21 @@ export const route: Route = {
...
     const $ = load(res.data);
 
     const title = $('.title').text();
 
     // 获取标题、时间及链接
-    const items = [];
+    const items: DataItem[] = [];
     $('.notice > ul > li > a').each((i, elem) => {
         items.push({
             title: $(elem.children[0]).text(),
```
Comment (TonyRL) [inline lib/routes/swpu/bgw.ts:49]: Please update it to use [map()](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Array/map) as well.

File: lib/routes/swpu/bgw.ts [hunk 4]
```diff
@@ -53,29 +53,28 @@ async function handler(ctx) {
...
+            async (item: DataItem) =>
+                (await cache.tryGet(item.link!, async () => {
+                    const resp = await got.get(item.link);
+                    const $ = load(resp.data);
+                    if ($('title').text().startsWith('系统提示')) {
+                        item.author = '系统';
+                        item.description = '无权访问';
+                    } else {
+                        item.author = '办公网';
+                        item.description = $('.v_news_content').html()?.toString();
```
[the remark this answers is not in the document]
Comment (CYTMWIA) [reply lib/routes/swpu/bgw.ts:66]: Type of `item.description` is `string | undefined`, but `.html()` returns `string | null`, which would give error ts(2322).
`?.toString()` will convert `null` to `undefined` to satisfy TypeScript.
Maybe I should write a function to do this?

[the remark this answers is not in the document]
Comment (CYTMWIA) [reply lib/routes/swpu/bgw.ts:66]: I decided to use `.html()!`.

**Pull request merged**